From 24ca0e4fad27f9fe83b6ec97c28e6c86373f3d8b Mon Sep 17 00:00:00 2001
From: Jochen Friedrich <jochen@scram.de>
Date: Fri, 3 May 2019 14:52:30 -0700
Subject: [PATCH 4/6] Clean up comiler warnings (Closes: #226561)

---
 src/toast.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/toast.c b/src/toast.c
index 05cd0b3..a779b9f 100644
--- a/src/toast.c
+++ b/src/toast.c
@@ -251,8 +251,8 @@ static char * emalloc P1((len), size_t len)
 {
 	char * s;
 	if (!(s = malloc(len))) {
-		fprintf(stderr, "%s: failed to malloc %d bytes -- abort\n",
-			progname, (int)len);
+		fprintf(stderr, "%s: failed to malloc %ld bytes -- abort\n",
+			progname, (long)len);
 		onintr();
 		exit(1);
 	}
@@ -386,7 +386,7 @@ static void update_times P0()
 		ut[0] = instat.st_atime;
 		ut[1] = instat.st_mtime;
 
-		(void) utime(outname, ut);
+		(void) utime(outname, (struct utimbuf *)ut);
 
 #endif	/* UTIMBUF */
 	}
@@ -586,8 +586,8 @@ static int process_decode P0()
 
 		if (cc != sizeof(s)) {
 			if (cc >= 0) fprintf(stderr,
-			"%s: incomplete frame (%d byte%s missing) from %s\n",
-					progname, (int)(sizeof(s) - cc),
+			"%s: incomplete frame (%ld byte%s missing) from %s\n",
+					progname, (long)(sizeof(s) - cc),
 					"s" + (sizeof(s) - cc == 1),
 					inname ? inname : "stdin" );
 			gsm_destroy(r);
-- 
2.21.0

