From 2a54a376f03a7b4fd5d77a4f345a139dc22cfb2c Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Fri, 3 May 2019 15:52:11 -0700
Subject: [PATCH 6/6] Fix installation destinations

---
 Makefile | 27 ++++++++++++++++++++-------
 1 file changed, 20 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index ad1f34e..a2e188b 100644
--- a/Makefile
+++ b/Makefile
@@ -79,9 +79,9 @@ INSTALL_ROOT	=
 # this directory.
 
 GSM_INSTALL_ROOT = $(INSTALL_ROOT)
-GSM_INSTALL_LIB = $(GSM_INSTALL_ROOT)/lib
-GSM_INSTALL_INC = $(GSM_INSTALL_ROOT)/inc
-GSM_INSTALL_MAN = $(GSM_INSTALL_ROOT)/man/man3
+GSM_INSTALL_LIB = $(GSM_INSTALL_ROOT)/usr/lib64
+GSM_INSTALL_INC = $(GSM_INSTALL_ROOT)/usr/include
+GSM_INSTALL_MAN = $(GSM_INSTALL_ROOT)/usr/share/man/man3
 
 
 # Where do you want to install the toast binaries and their manpage?
@@ -90,8 +90,8 @@ GSM_INSTALL_MAN = $(GSM_INSTALL_ROOT)/man/man3
 # of this directory.
 
 TOAST_INSTALL_ROOT	  = $(INSTALL_ROOT)
-TOAST_INSTALL_BIN = $(TOAST_INSTALL_ROOT)/bin
-TOAST_INSTALL_MAN = $(TOAST_INSTALL_ROOT)/man/man1
+TOAST_INSTALL_BIN = $(TOAST_INSTALL_ROOT)/usr/bin
+TOAST_INSTALL_MAN = $(TOAST_INSTALL_ROOT)/usr/share/man/man1
 
 #  Other tools
 
@@ -259,6 +259,9 @@ STUFF = 	ChangeLog			\
 
 GSM_INSTALL_TARGETS =	\
 		$(GSM_INSTALL_LIB)/libgsm.a		\
+		$(GSM_INSTALL_LIB)/libgsm.so.1.0.18	\
+		$(GSM_INSTALL_LIB)/libgsm.so.1		\
+		$(GSM_INSTALL_LIB)/libgsm.so		\
 		$(GSM_INSTALL_INC)/gsm.h		\
 		$(GSM_INSTALL_MAN)/gsm.3		\
 		$(GSM_INSTALL_MAN)/gsm_explode.3	\
@@ -357,51 +360,61 @@ toastuninstall:
 		fi
 
 $(TOAST_INSTALL_BIN)/toast:	$(TOAST)
+		mkdir -p $(TOAST_INSTALL_BIN)
 		-rm $@
 		cp $(TOAST) $@
 		chmod 755 $@
 
 $(TOAST_INSTALL_BIN)/untoast:	$(TOAST_INSTALL_BIN)/toast
+		mkdir -p $(TOAST_INSTALL_BIN)
 		-rm $@
 		ln $? $@
 
 $(TOAST_INSTALL_BIN)/tcat:	$(TOAST_INSTALL_BIN)/toast
+		mkdir -p $(TOAST_INSTALL_BIN)
 		-rm $@
 		ln $? $@
 
 $(TOAST_INSTALL_MAN)/toast.1:	$(MAN)/toast.1
+		mkdir -p $(TOAST_INSTALL_MAN)
 		-rm $@
 		cp $? $@
 		chmod 444 $@
 
 $(GSM_INSTALL_MAN)/gsm.3:	$(MAN)/gsm.3
+		mkdir -p $(GSM_INSTALL_MAN)
 		-rm $@
 		cp $? $@
 		chmod 444 $@
 
 $(GSM_INSTALL_MAN)/gsm_option.3:	$(MAN)/gsm_option.3
+		mkdir -p $(GSM_INSTALL_MAN)
 		-rm $@
 		cp $? $@
 		chmod 444 $@
 
 $(GSM_INSTALL_MAN)/gsm_explode.3:	$(MAN)/gsm_explode.3
+		mkdir -p $(GSM_INSTALL_MAN)
 		-rm $@
 		cp $? $@
 		chmod 444 $@
 
 $(GSM_INSTALL_MAN)/gsm_print.3:	$(MAN)/gsm_print.3
+		mkdir -p $(GSM_INSTALL_MAN)
 		-rm $@
 		cp $? $@
 		chmod 444 $@
 
 $(GSM_INSTALL_INC)/gsm.h:	$(INC)/gsm.h
+		mkdir -p $(GSM_INSTALL_INC)
 		-rm $@
 		cp $? $@
 		chmod 444 $@
 
-$(GSM_INSTALL_LIB)/libgsm.a:	$(LIBGSM)
+$(GSM_INSTALL_LIB)/libgsm.%:	$(LIB)/libgsm.%
+		mkdir -p $(GSM_INSTALL_LIB)
 		-rm $@
-		cp $? $@
+		cp -d $? $@
 		chmod 444 $@
 
 
-- 
2.21.0

